### PR TITLE
Feature release memory

### PR DESCRIPTION
**Pull request by sujata-m**
- Fixed memory consumption by releasing some memory using gc

**OSM to OSW**
_Without Fix:_ 

```
Line #    Mem usage    Increment  Occurrences   Line Contents
=============================================================
    15     90.4 MiB     90.4 MiB           1       @profile
    16                                             def convert(self) -> Response:
    17     90.4 MiB      0.0 MiB           1           try:
    18     91.6 MiB      1.2 MiB           1               unzipped_files = OSWHelper.unzip(self.zip_path, self.workdir)
    19     56.0 MiB    -35.6 MiB           1               input_file = OSWHelper.merge(osm_files=unzipped_files, output=self.workdir, prefix=self.prefix)
    20     56.5 MiB      0.5 MiB           1               output_file = Path(self.workdir, f'{self.prefix}.graph.osm.xml')
    21                                         
    22                                                     # Create the translation object.
    23     56.5 MiB      0.1 MiB           1               translation_object = OSMNormalizer()
    24                                         
    25                                                     # Create the ogr datasource
    26     57.4 MiB      0.9 MiB           1               datasource = ogr2osm.OgrDatasource(translation_object)
    27    426.1 MiB    368.7 MiB           1               datasource.open_datasource(input_file)
    28                                         
    29                                                     # Instantiate the ogr to osm converter class ogr2osm. OsmData and start the conversion process
    30    426.4 MiB      0.2 MiB           1               osm_data = ogr2osm.OsmData(translation_object)
    31   2243.2 MiB   1816.9 MiB           1               osm_data.process(datasource)
    32                                         
    33                                                     # Instantiate either ogr2osm.OsmDataWriter or ogr2osm.PbfDataWriter
    34   2243.2 MiB      0.0 MiB           1               data_writer = ogr2osm.OsmDataWriter(output_file, suppress_empty_tags=True)
    35   1724.8 MiB   -518.4 MiB           1               osm_data.output(data_writer)
    36                                         
    37   1724.9 MiB      0.1 MiB           1               data_writer.close()
    38                                                     # Delete merge file
    39   1725.8 MiB      0.9 MiB           1               Path(input_file).unlink()
    40   1725.9 MiB      0.1 MiB           1               resp = Response(status=True, generated_files=str(output_file))
    41                                                 except Exception as error:
    42                                                     print(error)
    43                                                     resp = Response(status=False, error=str(error))
    44   1725.9 MiB      0.0 MiB           1           return resp
```

_With Fix:_ 

```
Line #    Mem usage    Increment  Occurrences   Line Contents
=============================================================
    16     90.8 MiB     90.8 MiB           1       @profile
    17                                             def convert(self) -> Response:
    18     90.8 MiB      0.0 MiB           1           try:
    19     91.1 MiB      0.3 MiB           1               unzipped_files = OSWHelper.unzip(self.zip_path, self.workdir)
    20     56.3 MiB    -34.8 MiB           1               input_file = OSWHelper.merge(osm_files=unzipped_files, output=self.workdir, prefix=self.prefix)
    21     56.8 MiB      0.5 MiB           1               output_file = Path(self.workdir, f'{self.prefix}.graph.osm.xml')
    22                                         
    23                                                     # Create the translation object.
    24     56.9 MiB      0.1 MiB           1               translation_object = OSMNormalizer()
    25                                         
    26                                                     # Create the ogr datasource
    27     57.7 MiB      0.9 MiB           1               datasource = ogr2osm.OgrDatasource(translation_object)
    28    427.0 MiB    369.2 MiB           1               datasource.open_datasource(input_file)
    29                                         
    30                                                     # Instantiate the ogr to osm converter class ogr2osm. OsmData and start the conversion process
    31    427.2 MiB      0.2 MiB           1               osm_data = ogr2osm.OsmData(translation_object)
    32   2260.9 MiB   1833.6 MiB           1               osm_data.process(datasource)
    33                                         
    34   2329.4 MiB     68.6 MiB           1               del datasource
    35   2517.8 MiB    188.4 MiB           1               gc.collect()
    36                                         
    37                                                     # Instantiate either ogr2osm.OsmDataWriter or ogr2osm.PbfDataWriter
    38   2517.8 MiB      0.0 MiB           1               data_writer = ogr2osm.OsmDataWriter(output_file, suppress_empty_tags=True)
    39   1773.8 MiB   -744.0 MiB           1               osm_data.output(data_writer)
    40                                         
    41   2130.0 MiB    356.1 MiB           1               del osm_data
    42   2130.0 MiB      0.0 MiB           1               del data_writer
    43    128.8 MiB  -2001.1 MiB           1               gc.collect()
    44                                                     # Delete merge file
    45    129.0 MiB      0.2 MiB           1               Path(input_file).unlink()
    46    129.0 MiB      0.0 MiB           1               resp = Response(status=True, generated_files=str(output_file))
    47                                                 except Exception as error:
    48                                                     print(error)
    49                                                     resp = Response(status=False, error=str(error))
    50    129.0 MiB      0.0 MiB           1           return resp
```

**OSW to OSM**

_Without Fix:_

```
Line #    Mem usage    Increment  Occurrences   Line Contents
=============================================================
    18     90.3 MiB     90.3 MiB           1       @profile()
    19                                             async def convert(self) -> Response:
    20     90.3 MiB      0.0 MiB           1           try:
    21     90.3 MiB      0.0 MiB           1               print('Estimating number of ways, nodes, points, lines, zones and polygons in datasets...')
    22     90.3 MiB      0.0 MiB           1               tasks = [
    23     90.3 MiB      0.0 MiB           1                   OSWHelper.count_entities(self.osm_file_path, WayCounter),
    24     90.3 MiB      0.0 MiB           1                   OSWHelper.count_entities(self.osm_file_path, NodeCounter),
    25     90.3 MiB      0.0 MiB           1                   OSWHelper.count_entities(self.osm_file_path, PointCounter),
    26     90.3 MiB      0.0 MiB           1                   OSWHelper.count_entities(self.osm_file_path, LineCounter),
    27     90.3 MiB      0.0 MiB           1                   OSWHelper.count_entities(self.osm_file_path, ZoneCounter),
    28     90.3 MiB      0.0 MiB           1                   OSWHelper.count_entities(self.osm_file_path, PolygonCounter)
    29                                                     ]
    30                                         
    31    368.5 MiB    278.2 MiB           2               count_results = await asyncio.gather(*tasks)
    32                                         
    33    368.6 MiB      0.2 MiB           1               print('Creating networks from region extracts...')
    34    368.7 MiB      0.0 MiB           1               tasks = [OSWHelper.get_osm_graph(self.osm_file_path)]
    35   1194.1 MiB    825.4 MiB           2               osm_graph_results = await asyncio.gather(*tasks)
    36   1194.1 MiB      0.0 MiB           1               osm_graph_results = list(osm_graph_results)
    37   1194.1 MiB      0.0 MiB           1               OG = osm_graph_results[0]
    38                                         
    39   1197.8 MiB      3.7 MiB           2               await OSWHelper.simplify_og(OG)
    40                                         
    41   1301.0 MiB    103.2 MiB           2               await OSWHelper.construct_geometries(OG)
    42                                         
    43                                                     # for OG in osm_graph_results:
    44   1691.6 MiB    390.5 MiB           2               generated_files = await OSWHelper.write_og(self.workdir, self.filename, OG)
    45                                         
    46   1691.6 MiB      0.0 MiB           1               print(f'Created OSW files!')
    47   1691.6 MiB      0.0 MiB           1               self.generated_files = generated_files
    48   1691.6 MiB      0.0 MiB           1               resp = Response(status=True, generated_files=generated_files)
    49                                                 except Exception as error:
    50                                                     traceback.print_exc()
    51                                                     print(error)
    52                                                     resp = Response(status=False, error=str(error))
    53   1691.6 MiB      0.0 MiB           1           return resp
```

_With Fix:_

```
Line #    Mem usage    Increment  Occurrences   Line Contents
=============================================================
    22     90.3 MiB     90.3 MiB           1       @profile()
    23                                             async def convert(self) -> Response:
    24     90.3 MiB      0.0 MiB           1           try:
    25     90.3 MiB      0.0 MiB           1               print('Creating networks from region extracts...')
    26     90.3 MiB      0.0 MiB           1               tasks = [OSWHelper.get_osm_graph(self.osm_file_path)]
    27   1301.2 MiB   1210.9 MiB           2               osm_graph_results = await asyncio.gather(*tasks)
    28   1301.2 MiB      0.0 MiB           1               del tasks
    29   1288.7 MiB    -12.5 MiB           1               gc.collect()
    30                                         
    31   1288.7 MiB      0.0 MiB           1               osm_graph_results = list(osm_graph_results)
    32   1288.7 MiB      0.0 MiB           1               OG = osm_graph_results[0]
    33   1288.7 MiB      0.0 MiB           1               del osm_graph_results
    34                                         
    35   1297.8 MiB      9.0 MiB           2               await OSWHelper.simplify_og(OG)
    36   1297.8 MiB    -32.2 MiB           2               await OSWHelper.construct_geometries(OG)
    37                                         
    38                                                     # for OG in osm_graph_results:
    39   1335.5 MiB     37.7 MiB           2               generated_files = await OSWHelper.write_og(self.workdir, self.filename, OG)
    40   1335.5 MiB      0.0 MiB           1               del OG
    41                                         
    42   1335.5 MiB      0.0 MiB           1               print(f'Created OSW files!')
    43   1335.5 MiB      0.0 MiB           1               self.generated_files = generated_files
    44   1335.5 MiB      0.0 MiB           1               del generated_files
    45   1335.5 MiB      0.0 MiB           1               resp = Response(status=True, generated_files=self.generated_files)
    46                                                 except Exception as error:
    47                                                     traceback.print_exc()
    48                                                     print(error)
    49                                                     resp = Response(status=False, error=str(error))
    50                                                 finally:
    51    455.6 MiB   -880.0 MiB           1               gc.collect()
    52    455.6 MiB      0.0 MiB           1           return resp
```

### CODE REVIEW

File: src/osm_osw_reformatter/osm2osw/osm2osw.py [hunk 1]
```diff
@@ -16,36 +17,30 @@ def __init__(self, prefix: str, osm_file=None, workdir=None):
 
     async def convert(self) -> Response:
         try:
-            print('Estimating number of ways, nodes, points, lines, zones and polygons in datasets...')
```
Comment (susrisha) [inline src/osm_osw_reformatter/osm2osw/osm2osw.py:19]: Why is this deleted?

Comment (sujata-m) [reply src/osm_osw_reformatter/osm2osw/osm2osw.py:19]: That is not need,  we are calculating it but not using anywhere

**Review by susrisha — APPROVED** [review]
